source/precalculus/exercises/outcomes/FN/FN3/generator.sage

```diff
@@ -80,7 +75,16 @@ class Generator(BaseGenerator):
   @provide_data
   def graphics(data):
   # updated by clontz
+      plots=[]
+      for i in [0,1]:
+        plots.append( plot(data["graphs"][i]["pieces"][0],xmin=data["graphs"][i]["domain"][0],xmax=data["graphs"][i]["cut_points"][0][0],thickness=3) +
+                      plot(data["graphs"][i]["pieces"][1],xmin=data["graphs"][i]["cut_points"][0][0],xmax=data["graphs"][i]["cut_points"][1][0],thickness=3) +
+                      plot(data["graphs"][i]["pieces"][2],xmin=data["graphs"][i]["cut_points"][1][0],xmax=data["graphs"][i]["cut_points"][2][0],thickness=3) +
+                      plot(data["graphs"][i]["pieces"][3],xmin=data["graphs"][i]["cut_points"][2][0],xmax=data["graphs"][i]["cut_points"][3][0],thickness=3) +
+                      plot(data["graphs"][i]["pieces"][4],xmin=data["graphs"][i]["cut_points"][3][0],xmax=data["graphs"][i]["domain"][1],thickness=3)
+                    )
```

Comment:
Should have caught this in my review! Yes, all plot stuff needs to happen on this side. (I should probably raise an error when returning graphics objects on the data side...)